File: 1672 Richest Customer Wealth/go/main.go
package main

import "fmt"

func main() {
	fmt.Println(maximumWealth([][]int{{1, 2, 3}, {3, 2, 1}}))
	fmt.Println(maximumWealth([][]int{{1, 5}, {7, 3}, {3, 5}}))
	fmt.Println(maximumWealth([][]int{{2, 8, 7}, {7, 1, 3}, {1, 9, 5}}))
}

func maximumWealth(accounts [][]int) int {
	maxVal := 0
	for i := 0; i < len(accounts); i++ {
		val := 0

		for j := 0; j < len(accounts[i]); j++ {
			val += accounts[i][j]
		}

		if maxVal < val {
			maxVal = val
		}
	}

	return maxVal
}

File: 1672 Richest Customer Wealth/rust/main.rs
use std::time::Instant;

fn main() {
    let start = Instant::now();
    for _ in 0..10_000_000 {
        Solution::maximum_wealth(vec![vec![1, 2, 3], vec![3, 2, 1]]);
    }
    let test = vec!["Test", "Test"];

    println!("{:?}", start.elapsed());
}

struct Solution {}

impl Solution {
    pub fn maximum_wealth(accounts: Vec<Vec<i32>>) -> i32 {
        let mut max_val = 0;

        for i in 0..accounts.len() {
            let mut val = 0;
            for j in 0..accounts[i].len() {
                val += accounts[i][j];
            }

            if max_val < val {
                max_val = val;
            }
        }

        max_val
    }
}
